elastic/kibana
# [Rule Registry] Updating `index.mapping.total_fields.limit` index setting on concrete backing indices

## Summary

Upgrade testing on the QA cluster revealed that [this PR](https://github.com/elastic/kibana/pull/140598) to add the ECS component template to the observability alerts as data index templates will cause errors during rule registry initialization when the mappings are updated for existing concrete indices for 7.17 that have a total field limit of 1000. While we specify a higher limit in the index template specification, those changes are not applied to existing indices, instead they will apply when the indices roll over. This PR updates this specific index setting for existing concrete indices.

## To Verify

1. Run 7.17 and create an observability rule. Make sure that rule becomes active and writes out alerts as data (this ensures that the concrete index is created. Go to the index settings for this concrete index and verify that the total fields limit is 1000.
2. Run this branch using the same ES data and verify there are no rule registry initialization errors during startup. Go to the index settings for the concrete index and verify that the total fields limit has been updated to 1900.
